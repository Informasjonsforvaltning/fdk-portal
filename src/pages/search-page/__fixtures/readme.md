datasetsApiResponse.json is generated as:
curl "https://fellesdatakatalog.brreg.no/datasets?q=Marka&size=50" -H "Accept: application/json" -o applications/search/src/pages/search-page/__fixtures/datasetsApiResponse.json

todo - instead of readme, each json could have corresponding .sh script to update. 