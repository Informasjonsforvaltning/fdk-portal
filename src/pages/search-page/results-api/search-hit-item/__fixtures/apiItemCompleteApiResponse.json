{
  "id": "24fa3668-6ada-4a50-804e-1396f8470a24",
  "uri": "https://oidc-ver2.difi.no/kontaktinfo-oauth2-server/rest/apis",
  "title": {
    "no": "REST-API for Kontakt- og reserversjonsregisteret (KRR)"
  },
  "description": {
    "no": "Dette er eit REST-API for å tilby funksjonalitet tilsvarende dagens SOAP-baserte oppslagstjeneste for KRR.\n\nTilgang til API-et styres med OAuth2-baserte access_tokens. Desse vert utstedt frå ID-porten si OAuth2-teneste.\n\nLes meir på https://difi.github.io/idporten-oidc-dokumentasjon/oidc_api_krr_user.html\n"
  },
  "publisher": {
    "uri": "http://data.brreg.no/enhetsregisteret/enhet/974760673",
    "id": "974760673",
    "name": "REGISTERENHETEN I BRØNNØYSUND",
    "orgPath": "/STAT/912660680/974760673",
    "prefLabel": {
      "no": "Brønnøysundregistrene"
    }
  },
  "formats": [
    "application/json",
    "application/jose",
    "*/*"
  ],
  "openApi": {
    "openapi": "3.0.1",
    "info": {
      "title": "REST-API for Kontakt- og reserversjonsregisteret (KRR)",
      "description": "Dette er eit REST-API for å tilby funksjonalitet tilsvarende dagens SOAP-baserte oppslagstjeneste for KRR.\n\nTilgang til API-et styres med OAuth2-baserte access_tokens. Desse vert utstedt frå ID-porten si OAuth2-teneste.\n\nLes meir på https://difi.github.io/idporten-oidc-dokumentasjon/oidc_api_krr_user.html\n",
      "extensions": {}
    },
    "servers": [
      {
        "url": "//oidc-ver2.difi.no/kontaktinfo-oauth2-server/rest"
      }
    ],
    "tags": [
      {
        "name": "Person",
        "description": "Tilgang til data om innlogga brukar (brukaren knytt til access token).",
        "extensions": {}
      },
      {
        "name": "Personer",
        "description": "Hent ut opptil 1000 personar frå KRR pr. API-kall",
        "extensions": {}
      }
    ],
    "paths": {
      "/v1/person": {
        "get": {
          "tags": [
            "Person"
          ],
          "summary": "Hent KRR-data for brukaren knytt til access token",
          "description": "Minimumskrav til scope: user/kontaktinformasjon.read, user/spraak.read eller user/spraak.write. For å sjå kva data som blir gjeve ut basert på ulike scope, sjå https://difi.github.io/idporten-oidc-dokumentasjon/oidc_api_krr_user.html#bruk-av-oauth2",
          "operationId": "retrievePersonUsingGET",
          "parameters": [
            {
              "name": "Authorization",
              "in": "header",
              "description": "Authorization",
              "schema": {
                "type": "string",
                "extensions": {}
              },
              "extensions": {}
            }
          ],
          "responses": {
            "200": {
              "description": "Ok",
              "content": {
                "application/json": {
                  "schema": {
                    "$ref": "#/components/schemas/PersonResource",
                    "extensions": {}
                  }
                },
                "application/jose": {
                  "schema": {
                    "$ref": "#/components/schemas/PersonResource",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "401": {
              "description": "Invalid authorization token provided",
              "headers": {
                "WWW-Authenticate": {
                  "description": "error=\"invalid_token\" (not present if authorization header is missing or has the wrong type) and may contain error_description=\"...\" with further details. OAuth2-related errors - see https://tools.ietf.org/html/rfc6750#page-9",
                  "schema": {
                    "type": "string",
                    "extensions": {}
                  }
                }
              },
              "content": {
                "application/json": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                },
                "application/jose": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "403": {
              "description": "Forbidden, required scopes not associated with provided token",
              "headers": {
                "WWW-Authenticate": {
                  "description": "error=\"insufficient_scope\" and may contain error_description=\"...\" and scope=\"...\". OAuth2-related errors - see https://tools.ietf.org/html/rfc6750#page-9",
                  "schema": {
                    "type": "string",
                    "extensions": {}
                  }
                }
              },
              "content": {
                "application/json": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                },
                "application/jose": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            }
          },
          "security": [
            {
              "OAuth2": []
            }
          ],
          "extensions": {}
        },
        "patch": {
          "tags": [
            "Person"
          ],
          "summary": "Oppdater brukaren sin språkpreferanse",
          "description": "Krev scope user/spraak.write. Kun for å endre språkpreferanse.",
          "operationId": "patchUserUsingPATCH",
          "parameters": [
            {
              "name": "Authorization",
              "in": "header",
              "description": "Authorization",
              "schema": {
                "type": "string",
                "extensions": {}
              },
              "extensions": {}
            }
          ],
          "requestBody": {
            "description": "userResourcePatch",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/person patch document"
                }
              }
            },
            "required": true
          },
          "responses": {
            "204": {
              "description": "Query ok, no content in response",
              "content": {
                "*/*": {
                  "schema": {
                    "type": "object",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "400": {
              "description": "Malformed patch document. Example: empty request body, if unrecognized properties exists in JSON request body, missing spraak-parameter, invalid person-identificator.",
              "headers": {
                "WWW-Authenticate": {
                  "description": "Header present in case of Oauth2-related errors. Will have multiple values, among them: error=\"invalid_request\" and may contain error_description=\"...\" with further details. OAuth2-related errors - see https://tools.ietf.org/html/rfc6750#page-9",
                  "schema": {
                    "type": "string",
                    "extensions": {}
                  }
                }
              },
              "content": {
                "*/*": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "401": {
              "description": "Invalid authorization token provided",
              "headers": {
                "WWW-Authenticate": {
                  "description": "error=\"invalid_token\" (not present if authorization header is missing or has the wrong type) and may contain error_description=\"...\" with further details. OAuth2-related errors - see https://tools.ietf.org/html/rfc6750#page-9",
                  "schema": {
                    "type": "string",
                    "extensions": {}
                  }
                }
              },
              "content": {
                "*/*": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "403": {
              "description": "Forbidden, required scopes not associated with provided token",
              "headers": {
                "WWW-Authenticate": {
                  "description": "error=\"insufficient_scope\" and may contain error_description=\"...\" and scope=\"...\". OAuth2-related errors - see https://tools.ietf.org/html/rfc6750#page-9",
                  "schema": {
                    "type": "string",
                    "extensions": {}
                  }
                }
              },
              "content": {
                "*/*": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "415": {
              "description": "Unsupported patch document. Request has the wrong content type.",
              "content": {
                "*/*": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "422": {
              "description": "Unprocessable request. For example wrong value for preferredLanguage",
              "content": {
                "*/*": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "500": {
              "description": "Internal error.",
              "content": {
                "*/*": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            }
          },
          "security": [
            {
              "OAuth2": []
            }
          ],
          "extensions": {}
        },
        "extensions": {}
      },
      "/v1/personer": {
        "post": {
          "tags": [
            "Personer"
          ],
          "summary": "Hent ut 1 til 1000 personar frå KRR",
          "description": "Minimumskrav til scope: global/kontaktinformasjon.read, global/spraak.read eller global/spraak.write. For å sjå kva data som blir gjeve ut basert på ulike scope, sjå https://difi.github.io/idporten-oidc-dokumentasjon/oidc_api_krr.html#tilgjenglige-scopes",
          "operationId": "hentPersonerUsingPOST",
          "parameters": [
            {
              "name": "present",
              "in": "query",
              "schema": {
                "type": "boolean",
                "extensions": {}
              },
              "extensions": {}
            }
          ],
          "requestBody": {
            "description": "request",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/HentPersonerRequest"
                }
              }
            },
            "required": true
          },
          "responses": {
            "200": {
              "description": "Ok",
              "content": {
                "application/json": {
                  "schema": {
                    "$ref": "#/components/schemas/HentPersonerResponse",
                    "extensions": {}
                  }
                },
                "application/jose": {
                  "schema": {
                    "$ref": "#/components/schemas/HentPersonerResponse",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "401": {
              "description": "Invalid authorization token provided",
              "headers": {
                "WWW-Authenticate": {
                  "description": "error=\"invalid_token\" (not present if authorization header is missing or has the wrong type) and may contain error_description=\"...\" with further details. OAuth2-related errors - see https://tools.ietf.org/html/rfc6750#page-9",
                  "schema": {
                    "type": "string",
                    "extensions": {}
                  }
                }
              },
              "content": {
                "application/json": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                },
                "application/jose": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            },
            "403": {
              "description": "Forbidden, required scopes not associated with provided token",
              "headers": {
                "WWW-Authenticate": {
                  "description": "error=\"insufficient_scope\" and may contain error_description=\"...\" and scope=\"...\". OAuth2-related errors - see https://tools.ietf.org/html/rfc6750#page-9",
                  "schema": {
                    "type": "string",
                    "extensions": {}
                  }
                }
              },
              "content": {
                "application/json": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                },
                "application/jose": {
                  "schema": {
                    "$ref": "#/components/schemas/errors",
                    "extensions": {}
                  }
                }
              },
              "extensions": {}
            }
          },
          "security": [
            {
              "OAuth2": []
            }
          ],
          "extensions": {}
        },
        "extensions": {}
      }
    },
    "components": {
      "schemas": {
        "person patch document": {
          "title": "person patch document",
          "required": [
            "spraak"
          ],
          "type": "object",
          "properties": {
            "spraak": {
              "type": "string",
              "description": "ISO 639-1. Use empty string to delete user preference.",
              "extensions": {},
              "enum": [
                "en",
                "se",
                "nb",
                "nn",
                "\"\""
              ]
            }
          }
        },
        "ContactInfoResource": {
          "title": "ContactInfoResource",
          "type": "object",
          "properties": {
            "epostadresse": {
              "type": "string",
              "extensions": {}
            },
            "epostadresse_oppdatert": {
              "type": "string",
              "extensions": {}
            },
            "epostadresse_sist_verifisert": {
              "type": "string",
              "extensions": {}
            },
            "mobiltelefonnummer": {
              "type": "string",
              "extensions": {}
            },
            "mobiltelefonnummer_oppdatert": {
              "type": "string",
              "extensions": {}
            },
            "mobiltelefonnummer_sist_verifisert": {
              "type": "string",
              "extensions": {}
            }
          }
        },
        "DigitalPostResource": {
          "title": "DigitalPostResource",
          "type": "object",
          "properties": {
            "postkasseadresse": {
              "type": "string",
              "extensions": {}
            },
            "postkasseleverandoeradresse": {
              "type": "string",
              "extensions": {}
            }
          }
        },
        "HentPersonerRequest": {
          "title": "HentPersonerRequest",
          "type": "object",
          "properties": {
            "personidentifikatorer": {
              "type": "array",
              "extensions": {}
            }
          }
        },
        "links": {
          "title": "links",
          "type": "object",
          "properties": {
            "about": {
              "type": "string",
              "description": "Link to read more about the problem.",
              "example": "https://http.cat/422",
              "extensions": {}
            }
          }
        },
        "source": {
          "title": "source",
          "type": "object",
          "properties": {
            "pointer": {
              "type": "string",
              "description": "JSON-pointer to relevant field in request-body.",
              "example": "/spraak",
              "extensions": {}
            }
          }
        },
        "HentPersonerResponse": {
          "title": "HentPersonerResponse",
          "type": "object",
          "properties": {
            "personer": {
              "type": "array",
              "extensions": {}
            }
          }
        },
        "error": {
          "title": "error",
          "type": "object",
          "properties": {
            "status": {
              "type": "string",
              "description": "HTTP status code",
              "example": "422",
              "extensions": {}
            },
            "title": {
              "type": "string",
              "description": "a short, human-readable summary of the problem",
              "example": "Invalid value",
              "extensions": {}
            },
            "detail": {
              "type": "string",
              "description": "a human-readable explanation specific to this occurrence of the problem",
              "example": "See documentation for list of possible values.",
              "extensions": {}
            },
            "source": {
              "$ref": "#/components/schemas/source",
              "extensions": {}
            },
            "links": {
              "$ref": "#/components/schemas/links",
              "extensions": {}
            }
          },
          "description": "an error object according to JSON API specification."
        },
        "PersonResource": {
          "title": "PersonResource",
          "type": "object",
          "properties": {
            "personidentifikator": {
              "type": "string",
              "extensions": {}
            },
            "reservasjon": {
              "type": "string",
              "extensions": {},
              "enum": [
                "JA",
                "NEI"
              ]
            },
            "status": {
              "type": "string",
              "extensions": {},
              "enum": [
                "AKTIV",
                "SLETTET",
                "IKKE_REGISTRERT"
              ]
            },
            "varslingsstatus": {
              "type": "string",
              "extensions": {},
              "enum": [
                "KAN_IKKE_VARSLES",
                "KAN_VARSLES"
              ]
            },
            "kontaktinformasjon": {
              "$ref": "#/components/schemas/ContactInfoResource",
              "extensions": {}
            },
            "digital_post": {
              "$ref": "#/components/schemas/DigitalPostResource",
              "extensions": {}
            },
            "sertifikat": {
              "type": "string",
              "extensions": {}
            },
            "spraak": {
              "type": "string",
              "extensions": {}
            },
            "spraak_oppdatert": {
              "type": "string",
              "extensions": {}
            }
          },
          "description": "Kva felt ein får ut i responsen er avhengig av kva scope ein har, sjå https://difi.github.io/idporten-oidc-dokumentasjon/oidc_api_krr.html#tilgjenglige-scopes"
        },
        "errors": {
          "title": "errors",
          "type": "object",
          "properties": {
            "errors": {
              "type": "array",
              "extensions": {}
            }
          },
          "description": "an error response conforming to JSON API spec.  http://jsonapi.org/format/#errors"
        }
      },
      "securitySchemes": {
        "OAuth2": {
          "type": "OAUTH2",
          "flows": {},
          "extensions": {}
        }
      }
    }
  }
}
