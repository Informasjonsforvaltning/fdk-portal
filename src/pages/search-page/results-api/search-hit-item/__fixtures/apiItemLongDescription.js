export const apiItemLongDescription = {
  id: '32690443-e926-4e52-9988-dbac45085102',
  uri: 'jar:file:/app.jar!/BOOT-INF/classes!/datakatalog-api.json',
  title: {
    no: 'National Data Directory Search API'
  },
  description: {
    no:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like)."
  },
  isFree: false,
  isOpenAccess: false,
  isOpenLicense: false,
  isNationalComponent: false
};
