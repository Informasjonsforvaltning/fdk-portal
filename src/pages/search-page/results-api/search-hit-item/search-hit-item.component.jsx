import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import _ from 'lodash';

import localization from '../../../../lib/localization';
import './search-hit-item.scss';
import { getTranslateText } from '../../../../lib/translateText';
import { SearchHitHeader } from '../../../../components/search-hit-header/search-hit-header.component';
import { AlertMessage } from '../../../../components/alert-message/alert-message.component';
import {
  iconIsFree,
  iconIsNotFree,
  iconIsNotOpenAccess,
  iconIsNotOpenLicense,
  iconIsOpenAccess,
  iconIsOpenLicense
} from '../../../../components/api-icons';
import { getFirstLineOfText } from '../../../../lib/stringUtils';

const renderHeaderLink = (item, publisher, publishers, referenceData) => {
  if (!item) {
    return null;
  }
  const { title } = item;
  const link = `/apis/${encodeURIComponent(item.id)}`;
  return (
    <header>
      <SearchHitHeader
        tag="h2"
        title={getTranslateText(title)}
        titleLink={link}
        publisherLabel={`${localization.provider}:`}
        publisher={publisher}
        publisherItems={publishers}
        nationalComponent={item.nationalComponent}
        statusCode={item.statusCode}
        referenceData={referenceData}
        detialsPage={false}
      />
    </header>
  );
};

const renderDescription = description => {
  if (!description) {
    return null;
  }

  return (
    <div className="fdk-text-size-medium">
      <div className="uu-invisible" aria-hidden="false">
        Beskrivelse av api
      </div>
      {getFirstLineOfText(getTranslateText(description))}
    </div>
  );
};

const renderExpiredOrDeprecatedVersion = item => {
  const statusCode = _.get(item, 'statusCode');
  const deprecationInfoExpirationDate = _.get(
    item,
    'deprecationInfoExpirationDate'
  );
  const deprecationInfoReplacedWithUrl = _.get(
    item,
    'deprecationInfoReplacedWithUrl'
  );
  if (statusCode === 'REMOVED') {
    return (
      <AlertMessage type="info">
        <span>{localization.statusRemoved} </span>
        {deprecationInfoReplacedWithUrl && (
          <a href={deprecationInfoReplacedWithUrl}>
            {localization.statusReplacedUrl}
          </a>
        )}
      </AlertMessage>
    );
  }
  if (statusCode === 'DEPRECATED') {
    return (
      <AlertMessage type="info">
        <span>
          {deprecationInfoExpirationDate &&
            localization.formatString(
              localization.statusDeprecated,
              localization.during,
              deprecationInfoExpirationDate.substring(0, 4)
            )}
          {!deprecationInfoExpirationDate &&
            localization.formatString(localization.statusDeprecated, '', '')}
        </span>

        {deprecationInfoReplacedWithUrl && (
          <a href={deprecationInfoReplacedWithUrl}>
            {localization.statusReplacedUrl}
          </a>
        )}
      </AlertMessage>
    );
  }
  return null;
};

const renderFormat = formats => {
  if (!formats || formats.length === 0) {
    return null;
  }

  const formatItems = formats =>
    formats.map((item, index) => (
      <span key={index}>
        {index > 0 ? ', ' : ''}
        {item}
      </span>
    ));

  return (
    <div className="mb-2">
      <strong>{localization.format}:&nbsp;</strong>
      {formatItems(formats)}
    </div>
  );
};

export const SearchHitItem = ({
  item,
  fadeInCounter,
  publishers,
  referenceData
}) => {
  const searchHitClass = cx('search-hit', {
    'fade-in-200': fadeInCounter === 0,
    'fade-in-300': fadeInCounter === 1,
    'fade-in-400': fadeInCounter === 2
  });

  const { isFree, isOpenAccess, isOpenLicense } = item;

  return (
    <article className={searchHitClass}>
      <span className="uu-invisible" aria-hidden="false">
        SÃ¸ketreff.
      </span>

      {renderHeaderLink(
        item,
        _.get(item, 'publisher'),
        publishers,
        referenceData
      )}

      {renderExpiredOrDeprecatedVersion(item)}

      {renderDescription(_.get(item, 'description'))}

      {renderFormat(_.get(item, 'formats'))}

      <div className="access-icons mt-2 d-flex justify-content-between">
        {isFree === true && iconIsFree()}
        {isFree === false && iconIsNotFree()}
        {isOpenAccess === true && iconIsOpenAccess()}
        {isOpenAccess === false && iconIsNotOpenAccess()}
        {isOpenLicense === true && iconIsOpenLicense()}
        {isOpenLicense === false && iconIsNotOpenLicense()}
      </div>
    </article>
  );
};

SearchHitItem.defaultProps = {
  referenceData: null,
  item: {},
  publishers: null
};

SearchHitItem.propTypes = {
  item: PropTypes.shape({}),
  publishers: PropTypes.object,
  referenceData: PropTypes.object
};
