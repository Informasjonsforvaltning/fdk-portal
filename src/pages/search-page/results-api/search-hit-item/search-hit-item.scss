@import '../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.search-hit {
  background-color: $color-white;
  border-radius: 5px;
  color: $color-neutral-darkest;
  padding: 1.5em 2em;
  margin-bottom: 1em;
  .search-hit__version {
    background-color: $color-primary-lighter;
    border-radius: 2px;
  }
}
[class^="icon2-"] {
    font-size: 34px;
    vertical-align: -9px;
    padding-right: 10px;
}
