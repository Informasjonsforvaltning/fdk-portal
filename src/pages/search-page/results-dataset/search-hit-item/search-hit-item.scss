@import '../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-a-search-hit {
  text-decoration: none;
}

.theme-fdk {
  .fdk-icon-public {
    background-image: url("/img/icon-dataset-public-md.svg");
  }
}

.theme-nap {
  .fdk-icon-public {
    background-image: url("/img/icon-dataset-public-nap-md.svg");
  }
}

.fdk-icon-public {
  background-image: url("/img/icon-dataset-public-md.svg");
  width: 2em; 
  height: 2em;
}

.fdk-icon-non-public {
  background-image: url("/img/icon-dataset-not-public-md.svg");
  width: 2em; 
  height: 2em;
}

.fdk-icon-restricted {
  background-image: url("/img/icon-dataset-restricted-md.svg");
  width: 2em; 
  height: 2em;
}

.fdk-color-restricted {
  color: $color-warning-darker !important;
}