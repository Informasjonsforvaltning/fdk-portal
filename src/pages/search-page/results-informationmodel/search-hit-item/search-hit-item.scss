@import '../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.search-hit {
  background-color: $color-white;
  border-radius: 5px;
  color: $color-neutral-darkest;
  padding: 1.5em 2em;
  margin-bottom: 1em;
}
