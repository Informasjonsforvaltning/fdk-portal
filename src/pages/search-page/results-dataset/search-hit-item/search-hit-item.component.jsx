import React from 'react';
import PropTypes from 'prop-types';
import * as _ from 'lodash';

import localization from '../../../../lib/localization';
import { getTranslateText } from '../../../../lib/translateText';
import { SearchHitHeader } from '../../../../components/search-hit-header/search-hit-header.component';
import './search-hit-item.scss';
import {
  getReferenceDataByUri,
  REFERENCEDATA_PATH_DISTRIBUTIONTYPE
} from '../../../../redux/modules/referenceData';

const renderFormats = (source, referenceData) => {
  const { distribution } = source;

  const children = distributions => {
    const nodes = [];
    distributions.forEach(item => {
      let formatString;
      const { format } = item;
      let { type } = item;
      if (
        type &&
        (type !== 'API' && type !== 'Feed' && type !== 'Nedlastbar fil')
      ) {
        const distributionType = getReferenceDataByUri(
          referenceData,
          REFERENCEDATA_PATH_DISTRIBUTIONTYPE,
          type
        );
        if (distributionType) {
          type = getTranslateText(distributionType.prefLabel);
        } else {
          type = null;
        }
      }
      if (format && typeof format !== 'undefined') {
        formatString = format.join(', ');
      }
      nodes.push(` ${type} (${formatString})`);
    });
    return nodes;
  };

  if (distribution && _.isArray(Object.keys(distribution))) {
    return (
      <div className="d-flex flex-wrap mb-4">
        {localization.search_hit.distributionType}:{children(distribution)}
      </div>
    );
  }
  return null;
};

const renderAccessRights = accessRight => {
  if (!accessRight) {
    return null;
  }

  const { code } = accessRight || {
    code: null
  };
  let accessRightsIconClass;
  let accessRightsColorClass;
  let accessRightsLabel;

  switch (code) {
    case 'NON_PUBLIC':
      accessRightsIconClass = 'fdk-icon-non-public';
      accessRightsColorClass = 'fdk-color-unntatt';
      accessRightsLabel =
        localization.dataset.accessRights.authorityCode.nonPublicDetailsLabel;
      break;
    case 'RESTRICTED':
      accessRightsIconClass = 'fdk-icon-restricted';
      accessRightsColorClass = 'fdk-color-restricted';
      accessRightsLabel =
        localization.dataset.accessRights.authorityCode.restrictedDetailsLabel;
      break;
    case 'PUBLIC':
      accessRightsIconClass = 'fdk-icon-public';
      accessRightsColorClass = 'fdk-color-offentlig';
      accessRightsLabel =
        localization.dataset.accessRights.authorityCode.publicDetailsLabel;
      break;
    default:
      accessRightsLabel = localization.unknown;
  }

  return (
    <div className="d-flex align-items-center mb-4">
      <i className={`mr-2 ${accessRightsIconClass}`} />
      <div className={accessRightsColorClass}>{accessRightsLabel}</div>
    </div>
  );
};

const renderSample = dataset => {
  const { sample } = dataset;
  if (Array.isArray(sample) && sample.length > 0) {
    return (
      <div className="mb-4" id="search-hit-sample">
        {localization.search_hit.sample}
      </div>
    );
  }
  return null;
};

export const SearchHitItem = props => {
  const { referenceData, dataset } = props;
  const { id, catalog, publisher, theme, provenance, accessRights } =
    dataset || {};
  let { title, description, objective } = dataset || {};
  title = getTranslateText(title);
  description = getTranslateText(description);
  objective = getTranslateText(objective);

  if (description && description.length > 220) {
    description = `${description.substr(0, 220)}...`;
  } else if (description && description.length < 150 && objective) {
    const freeLength = 200 - description.length;
    const objectiveLength = objective.length;
    description = `${description} ${objective.substr(0, 200 - freeLength)} ${
      objectiveLength > freeLength ? '...' : ''
    }`;
  }
  const link = `/datasets/${id}`;

  return (
    <article>
      <article
        className="fdk-a-search-hitx search-hit"
        title={`${localization.result.dataset}: ${title}`}
      >
        <span className="uu-invisible" aria-hidden="false">
          SÃ¸ketreff.
        </span>

        <header>
          <SearchHitHeader
            tag="h2"
            title={title}
            titleLink={link}
            catalog={catalog}
            publisherLabel={localization.search_hit.owned}
            publisher={publisher}
            theme={theme}
            nationalComponent={_.get(provenance, 'code') === 'NASJONAL'}
            referenceData={referenceData}
            darkThemeBackground={false}
          />
        </header>
        <p className="fdk-text-size-medium">
          <span className="uu-invisible" aria-hidden="false">
            Beskrivelse av datasettet,
          </span>
          {description}
        </p>

        {renderAccessRights(accessRights)}

        {renderFormats(dataset, referenceData)}

        {renderSample(dataset)}
      </article>
    </article>
  );
};

SearchHitItem.defaultProps = {
  dataset: null,
  referenceData: null
};

SearchHitItem.propTypes = {
  dataset: PropTypes.object,
  referenceData: PropTypes.object
};
