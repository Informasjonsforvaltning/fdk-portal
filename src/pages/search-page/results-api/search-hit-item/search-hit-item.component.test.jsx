import React from 'react';
import { shallow } from 'enzyme';
import { SearchHitItem } from './search-hit-item.component';
import apiItemComplete from './__fixtures/apiItemCompleteApiResponse.json';
import { apiItemLongDescription } from './__fixtures/apiItemLongDescription';

let defaultProps;

test('should render api-SearchHitItem correctly with no props', () => {
  const wrapper = shallow(<SearchHitItem />);
  expect(wrapper).toMatchSnapshot();
});

test('should render api-SearchHitItem correctly', () => {
  defaultProps = {
    item: apiItemComplete
  };
  const wrapper = shallow(<SearchHitItem {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render api-SearchHitItem with long description', () => {
  const wrapper = shallow(<SearchHitItem item={apiItemLongDescription} />);
  expect(wrapper).toMatchSnapshot();
});
