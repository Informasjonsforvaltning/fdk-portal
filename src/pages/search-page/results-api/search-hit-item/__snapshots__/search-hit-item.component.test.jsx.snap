// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render api-SearchHitItem correctly 1`] = `
<article
  className="search-hit"
>
  <span
    aria-hidden="false"
    className="uu-invisible"
  >
    Søketreff.
  </span>
  <header>
    <SearchHitHeader
      darkThemeBackground={false}
      detialsPage={false}
      externalLink={false}
      nationalComponent={false}
      publisher={
        Object {
          "id": "974760673",
          "name": "REGISTERENHETEN I BRØNNØYSUND",
          "orgPath": "/STAT/912660680/974760673",
          "prefLabel": Object {
            "no": "Brønnøysundregistrene",
          },
          "uri": "http://data.brreg.no/enhetsregisteret/enhet/974760673",
        }
      }
      publisherItems={null}
      publisherLabel="Provider:"
      publisherTag="span"
      referenceData={null}
      statusCode={null}
      tag="h2"
      theme={Array []}
      title="REST-API for Kontakt- og reserversjonsregisteret (KRR)"
      titleLink="/apis/24fa3668-6ada-4a50-804e-1396f8470a24"
    />
  </header>
  <div
    className="fdk-text-size-medium"
  >
    <div
      aria-hidden="false"
      className="uu-invisible"
    >
      Beskrivelse av api
    </div>
    Dette er eit REST-API for å tilby funksjonalitet tilsvarende dagens SOAP-baserte oppslagstjeneste for KRR. ...
  </div>
  <div
    className="mb-2"
  >
    <strong>
      Format
      : 
    </strong>
    <span
      key="0"
    >
      application/json
    </span>
    <span
      key="1"
    >
      , 
      application/jose
    </span>
    <span
      key="2"
    >
      , 
      */*
    </span>
  </div>
  <div
    className="access-icons mt-2 d-flex justify-content-between"
  />
</article>
`;

exports[`should render api-SearchHitItem correctly with no props 1`] = `
<article
  className="search-hit"
>
  <span
    aria-hidden="false"
    className="uu-invisible"
  >
    Søketreff.
  </span>
  <header>
    <SearchHitHeader
      darkThemeBackground={false}
      detialsPage={false}
      externalLink={false}
      nationalComponent={false}
      publisher={null}
      publisherItems={null}
      publisherLabel="Provider:"
      publisherTag="span"
      referenceData={null}
      statusCode={null}
      tag="h2"
      theme={Array []}
      title={null}
      titleLink="/apis/undefined"
    />
  </header>
  <div
    className="access-icons mt-2 d-flex justify-content-between"
  />
</article>
`;

exports[`should render api-SearchHitItem with long description 1`] = `
<article
  className="search-hit"
>
  <span
    aria-hidden="false"
    className="uu-invisible"
  >
    Søketreff.
  </span>
  <header>
    <SearchHitHeader
      darkThemeBackground={false}
      detialsPage={false}
      externalLink={false}
      nationalComponent={false}
      publisher={null}
      publisherItems={null}
      publisherLabel="Provider:"
      publisherTag="span"
      referenceData={null}
      statusCode={null}
      tag="h2"
      theme={Array []}
      title="National Data Directory Search API"
      titleLink="/apis/32690443-e926-4e52-9988-dbac45085102"
    />
  </header>
  <div
    className="fdk-text-size-medium"
  >
    <div
      aria-hidden="false"
      className="uu-invisible"
    >
      Beskrivelse av api
    </div>
    It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, ...
  </div>
  <div
    className="access-icons mt-2 d-flex justify-content-between"
  >
    <span
      className="access-icon flex-fill fdk-color-danger"
    >
      <span
        className="icon2-icon-api-cost"
      >
        <span
          className="path1"
        />
        <span
          className="path2"
        />
      </span>
      Not free
    </span>
    <span
      className="access-icon flex-fill fdk-color-danger"
    >
      <span
        className="icon2-icon-api-access-not-limited"
      >
        <span
          className="path1"
        />
        <span
          className="path2"
        />
      </span>
      Not public access
    </span>
    <span
      className="access-icon flex-fill fdk-color-danger"
    >
      <span
        className="icon2-icon-api-license"
      >
        <span
          className="path1"
        />
        <span
          className="path2"
        />
        <span
          className="path3"
        />
        <span
          className="path4"
        />
        <span
          className="path5"
        />
        <span
          className="path6"
        />
      </span>
      License not open
    </span>
  </div>
</article>
`;
